File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.java
package org.wordpress.android.ui;

import android.support.annotation.Nullable;

public enum JetpackConnectionSource {
    NOTIFICATIONS("notifications"),
    STATS("stats");

    private final String mValue;

    JetpackConnectionSource(String value) {
        mValue = value;
    }

    @Nullable
    public static JetpackConnectionSource fromString(String value) {
        if (NOTIFICATIONS.mValue.equals(value)) {
            return NOTIFICATIONS;
        } else if (STATS.mValue.equals(value)) {
            return STATS;
        } else {
            return null;
        }
    }

    @Override
    public String toString() {
        return mValue;
    }
}

File: WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionSource.kt
package org.wordpress.android.ui

enum class JetpackConnectionSource(private val value: String) {
    NOTIFICATIONS("notifications"),
    STATS("stats"),
    XMLRPC_DISABLED("xmlrpc_disabled");

    override fun toString() = value

    companion object {
        @JvmStatic
        fun fromString(value: String) = when {
            NOTIFICATIONS.value == value -> NOTIFICATIONS
            STATS.value == value -> STATS
            XMLRPC_DISABLED.value == value -> XMLRPC_DISABLED
            else -> null
        }
    }
}
